routes/index.js

```diff
@@ -42,10 +42,17 @@ router.get('/logout',
 // register new user with local strategy
 router.post('/register', (req, res) => {
   Student.register(new Student({ username: req.body.username }),
-    req.body.password, (regErr, account) => {
+    req.body.password, (regErr, newAccount) => {
       if (regErr) {
         console.error(regErr);
       }
+      // hotfix. Redux demands these fields be filled,
+      // but they have no default values
+      const account = Object.assign(newAccount, {
+        name: 'Test User',
+        entryYear: '2001',
+        major: 'Mechanical Engineering',
+      });
```

Comment:
The code above actually overwrites everything in `account` and `newAccount` with `name: 'Test User'` etc., even if `newAccount.name` has a value. Not sure if that's what you wanted.

To merge the objects with precedence to supplied `newAccount` properties, use:
`const account = Object.assign({name: 'Test User', entryYear: '2001', major: 'Mechanical Engineering'}, newAccount)` (if you keep the defaults as an object literal – in that case, the literal is mutated into the merged object each time)
or
`const account = Object.assign({}, AccountDefaults, newAccount)` (if you move the defaults to a global).

If you're using redux, you've probably got Babel / webpack set up with the object rest spread operator. In that case, you can also write
`const account = {...AccountDefaults, ...newAccount}`

<https://jsfiddle.net/owsteele/5wk3tzLf/1/>